pkg/segment/query/processor/statscommand.go

```diff
@@ -250,6 +282,10 @@ func (p *statsProcessor) processMeasureOperations(inputIQR *iqr.IQR) (*iqr.IQR,
 		}
 
 		for i := range values {
+			if !p.shouldContinueProcessing {
+				return nil, nil
+			}
```

Comment:
nit: is this necessary? I'd rather not include this in every processor command, and it seems like this would only rarely save CPU (because cancelling a query is probably rare, and the batch sizes aren't too huge)